src/controllers/historycontroller.js

```diff
@@ -0,0 +1,47 @@
+class HistoryController extends Controller
+{
+    static addEvent(event)
```

Comment:
Does this have to be a static method? I wrote a `registerEvent` method in the parent class to register jQuery event bindings, but I don't think what you are trying to achieve here involves events so much.

Perhaps just keep them as separate non-static functions?